daemon/algod/api/server/v2/handlers.go

```diff
@@ -1152,9 +1166,20 @@ func (v2 *Handlers) TealCompile(ctx echo.Context) error {
 	}
 	pd := logic.HashProgram(ops.Program)
 	addr := basics.Address(pd)
+
+	// If source map flag is enabled, then return the map.
+	var encodedMap *map[string]interface{}
+	if sourcemapFlag {
+		sourcemap = logic.GetSourceMap([]string{}, ops.OffsetToLine)
+		encodedMapString, _ := json.Marshal(sourcemap)
+		json.Unmarshal(encodedMapString, &encodedMap)
```

Comment:
This is to convert the `logic.SourceMap` object to a `map[string]interface{}`, right? If you're interested in messing with the type system a little to make this more efficient, what would be an improvement is creating a new type that embeds `generated.CompileResponse` and overrides the sourcemap field. Something like this maybe:

```golang
type CompileResponseWithSourceMap struct {
	generated.CompileResponse
        // note: the Go field name "Sourcemap" doesn't matter, only the declared json name
	Sourcemap logic.SourceMap `json:"sourcemap,omitempty"`
}
```

You can then pass this object to `ctx.JSON` instead of `generated.CompileResponse`.

Also, you're passing a `**map[string]interface{}` as the second argument to `json.Unmarshal`, does that work?

Reply:
This is neat, I changed it to the suggestion and it looks much cleaner.

I think `Unmarshal` needs to be supplied with a pointer to whatever object we want to store the data, and in this case it was the `**map[string]interface{}`. Looks weird, but it seemed to work in the unit tests.